app/org/sagebionetworks/bridge/validators/SurveyAnswerValidator.java

```diff
@@ -195,17 +197,52 @@ private void validateType(Errors errors, StringConstraints con, String answer) {
         } else if (con.getMaxLength() != null && answer.length() > con.getMaxLength()) {
             rejectField(errors, "constraints", "%s is longer than %s characters", answer, con.getMaxLength());
         }
-        if (StringUtils.isNotBlank(con.getPattern()) && !answer.matches(con.getPattern())) {
+        if (StringUtils.isNotBlank(con.getPattern()) && answer != null && !answer.matches(con.getPattern())) {
             rejectField(errors, "constraints", "%s does not match the regular expression /%s/", answer, con.getPattern());
         }
     }
 
-    private void validateType(Errors errors, DurationConstraints constraints, String answer) {
+    private void validateType(Errors errors, DurationConstraints con, String answer) {
+        double seconds = 0L;
         try {
-            // TODO: The docs don't say it throws an exception when wrong...
-            Period.parse(answer);
+            seconds = new DateTime(0L).plus(Period.parse(answer)).toDateTime().getMillis()/1000L;
         } catch(Throwable t) {
             rejectField(errors, "constraints", "%s is not a valid ISO 8601 duration string", answer);
+            return;
+        }
+        if (con.getUnit() != null) {
+            double value = convertPeriodToDurationUnits(con, seconds);
+            if (value > 0) {
```

Comment:
Should this be removed? the options are: declined, there's no validation, or it's within the range, we don't skip just because the answer is zero.